donation/views.py

```diff
@@ -51,16 +74,27 @@ def donation_list(request):
 
     return render(request, 'donation/list.html', context)
 
+
+@login_required(login_url='/admin/login')
```

Comment:
If you change this line to `@login_required(login_url='/admin/login/?next=/donation/create)` it should redirect to the creation form.